orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/SpinnakerMetadataServerGroupTagGenerator.java

```diff
@@ -59,11 +73,66 @@
       value.put("comments", (String) context.get("comments"));
     }
 
+    String cluster = null;
+    try {
+      cluster = Names.parseName(serverGroup).getCluster();
+
+      Map<String, Object> previousServerGroup = getPreviousServerGroup(
+        execution.getApplication(),
+        account,
+        cluster,
+        cloudProvider,
+        location
+      );
+
+      if (previousServerGroup != null) {
+        value.put("previousServerGroup", previousServerGroup);
+      }
+    } catch (Exception e) {
+      // failure to populate `previousServerGroup` is not considered a fatal error that would cause this task to fail
+      log.error("Unable to determine ancestor image details for {}:{}:{}:{}", cloudProvider, account, location, cluster, e);
+    }
+
     Map<String, Object> tag = new HashMap<>();
     tag.put("name", "spinnaker:metadata");
     tag.put("value", value);
 
     return Collections.singletonList(tag);
   }
 
+  Map<String, Object> getPreviousServerGroup(String application,
+                                             String account,
+                                             String cluster,
+                                             String cloudProvider,
+                                             String location) {
+    return retrySupport.retry(() -> {
+      try {
+        Map<String, Object> targetServerGroup = oortService.getServerGroupSummary(
+          application,
+          account,
+          cluster,
+          cloudProvider,
+          location,
+          "ANCESTOR",
+          "image",
+          "true"
+        );
+
+        Map<String, Object> previousServerGroup = new HashMap<>();
+        previousServerGroup.put("name", targetServerGroup.get("serverGroupName"));
+        previousServerGroup.put("imageName", targetServerGroup.get("imageName"));
+        previousServerGroup.put("imageId", targetServerGroup.get("imageId"));
```

Comment:
Haven't actually implemented the alternative rollback strategy, so may need to revisit this if more than `imageId` is needed.

Reply:
might as well include instance type, load balancers, security groups (not things that change much, but for the sake of completeness...)

Reply:
+1 to that

Reply:
Not sure I agree but could see how the naming of the key would make it seem more exhaustive than it is currently.

My inclination would be to leave as is (with only image details for now) rather than complicate the rollback process by possibly exposing load balancers, instance types, etc.

Going to let this chill for an hour or two and mull it over.